### PR TITLE
Update to RBAC/PIM Section

### PR DESCRIPTION
**Pull request by SamBH**
Update to RBAC/PIM section of the Platform Design in response to [issue #170](https://github.com/govau/desktop.gov.au/issues/170). 

Expanded on available PIM assignment scopes to include administrative units, applications, and service principals.

Reviewed by @oobedan.

### CODE REVIEW

File: blueprint/platform.md [hunk 1]
```diff
@@ -1665,26 +1665,26 @@ Microsoft 365 Compliance Center | Available from web console | The console is av
...
+* Approver - the Approver attribute specifies the person or people who can approve role activation requests.
+* Notification - the Notification attribute specifies that a pending request is awaiting approval via email.
+* Incident Request Ticket - the Incident Request Ticket attribute specifies that the approver add an incident ticket number to the approval request.
+* Multi-factor Authentication - the Multi-factor Authentication attribute specifies whether MFA is required for activation.
+
+Azure AD roles can be assigned via PIM to various scope types, depending on the specific role being assigned. The scope types include:
+
+* Directory - Roles that apply permissions across the entire Azure AD tenant.
+* Administrative unit - Configurable by administrators to segregate permissions within organisation into specific business units or locations.  Note only [specific Azure AD roles](https://docs.microsoft.com/en-au/azure/active-directory/roles/admin-units-assign-roles#roles-that-can-be-assigned-with-administrative-unit-scope) support can be assigned with administrative unit scope.
```
Comment (pl4nty) [inline blueprint/platform.md:1685]:
```suggestion
* Administrative unit - Configurable by administrators to segregate permissions within organisation into specific business units or locations.  Note only [specific Azure AD roles](https://docs.microsoft.com/en-au/azure/active-directory/roles/admin-units-assign-roles#roles-that-can-be-assigned-with-administrative-unit-scope) can be assigned with administrative unit scope.
```

File: blueprint/platform.md [hunk 2]
```diff
@@ -1665,26 +1665,26 @@ Microsoft 365 Compliance Center | Available from web console | The console is av
...
+* Application - A specific application registered to Azure AD. These are listed under [App registerations](https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredApps) in the Azure portal.
```
Comment (pl4nty) [inline blueprint/platform.md:1686]:
```suggestion
* Application - A specific application registered to Azure AD. These are listed under [App registrations](https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredApps) in the Azure portal.
```

**Review by pl4nty — COMMENTED** [review]
Thanks for the quick response to my feedback issue, really appreciate it! Just noticed some small things

**Comment by SamBH** [comment]
@zedzedzed I have committed the suggested edits and believe this PR can be approved unless you have any edits to add?

**Review by zedzedzed — APPROVED** [review]

**Pull request merged**